drivers/adc/adc_stm32f4.c

```diff
@@ -0,0 +1,501 @@
+	return rc;
+}
+
+static struct adc_driver_api api_funcs = {
+	.enable  = adc_stm32f4_enable,
+	.disable = adc_stm32f4_disable,
+	.read    = adc_stm32f4_read,
+};
+
+
+int adc_stm32f4_cfgChannel (ADC_HandleTypeDef * phadc, uint32_t c, uint32_t r) {
+	ADC_ChannelConfTypeDef cfg;
+	GPIO_InitTypeDef gpioCfg;
+	GPIO_TypeDef * GPIOport;
+
+	SYS_LOG_DBG("try to config adc channel %u (rank=%u) ...", c, r);
+
+	cfg.Rank = r;
+	cfg.SamplingTime = ADC_SAMPLETIME_480CYCLES;
+	cfg.Offset = 0;
+	
+	gpioCfg.Mode = GPIO_MODE_ANALOG;
+	gpioCfg.Pull = GPIO_NOPULL;
+
+	// set the pins and special function registers
+	GPIOport =  NULL;
+
+	switch (c) {
+		case adcChannel_0:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_0;
+			cfg.Channel = ADC_CHANNEL_0;
+			break;
+		case adcChannel_1:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_1;
+			cfg.Channel = ADC_CHANNEL_1;
+			break;
+		case adcChannel_2:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_2;
+			cfg.Channel = ADC_CHANNEL_2;
+			break;
+		case adcChannel_3:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_3;
+			cfg.Channel = ADC_CHANNEL_3;
+			break;
+		case adcChannel_4:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_4;
+			cfg.Channel = ADC_CHANNEL_4;
+			break;
+		case adcChannel_5:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_5;
+			cfg.Channel = ADC_CHANNEL_5;
+			break;
+		case adcChannel_6:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_6;
+			cfg.Channel = ADC_CHANNEL_6;
+			break;
+		case adcChannel_7:
+			GPIOport = GPIOA;
+			gpioCfg.Pin = GPIO_PIN_7;
+			cfg.Channel = ADC_CHANNEL_7;
+			break;
+		case adcChannel_8:
+			GPIOport = GPIOB;
+			gpioCfg.Pin = GPIO_PIN_0;
+			cfg.Channel = ADC_CHANNEL_8;
+			break;
+		case adcChannel_9:
+			GPIOport = GPIOB;
+			gpioCfg.Pin = GPIO_PIN_1;
+			cfg.Channel = ADC_CHANNEL_9;
+			break;
+		case adcChannel_10:
+			GPIOport = GPIOC;
+			gpioCfg.Pin = GPIO_PIN_0;
+			cfg.Channel = ADC_CHANNEL_10;
+			break;
+		case adcChannel_11:
+			GPIOport = GPIOC;
+			gpioCfg.Pin = GPIO_PIN_1;
+			cfg.Channel = ADC_CHANNEL_11;
+			break;
+		case adcChannel_12:
+			GPIOport = GPIOC;
+			gpioCfg.Pin = GPIO_PIN_2;
+			cfg.Channel = ADC_CHANNEL_12;
+			break;
+		case adcChannel_13:
+			GPIOport = GPIOC;
+			gpioCfg.Pin = GPIO_PIN_3;
+			cfg.Channel = ADC_CHANNEL_13;
+			break;
+		case adcChannel_14:
+			GPIOport = GPIOC;
+			gpioCfg.Pin = GPIO_PIN_4;
+			cfg.Channel = ADC_CHANNEL_14;
+			break;
+		case adcChannel_15:
+			GPIOport = GPIOC;
+			gpioCfg.Pin = GPIO_PIN_5;
+			cfg.Channel = ADC_CHANNEL_15;
+			break;
+
+		case adcChannel_vref:
+			cfg.Channel = ADC_CHANNEL_VREFINT;
+			ADC->CCR |= ADC_CCR_TSVREFE;
+			break;
+
+		case adcChannel_temp:
+			cfg.Channel = ADC_CHANNEL_TEMPSENSOR;
+			break;
+
+		case adcChannel_vbat:
+			cfg.Channel = ADC_CHANNEL_VBAT;
+			ADC->CCR |= ADC_CCR_VBATE;
+			break;
+
+		default:
+			// none
+			break;
+	}
+	if(GPIOport) HAL_GPIO_Init(GPIOport, &gpioCfg);
```

Comment:
In Zephyr base code, we enforce the use of braces, even for one liners.
This should looks like:
```
if(GPIOport) {
    HAL_GPIO_Init(GPIOport, &gpioCfg);
}
```